python/ray/dataframe/dataframe.py

```diff
@@ -1411,12 +1447,12 @@ def _shuffle(df, indices, chunksize):
     i = 0
     partition = []
     while len(indices) > chunksize:
-        oids = df.reindex(indices[:chunksize]).dropna()
+        oids = df.reindex(indices[:chunksize])
```

Comment:
This was the source of the `np.nan` transpose bug.

Reply:
What was the reason for the `dropna()` call before?

Reply:
To conserve space. If we didn't have any data on this partition, we are still sending some na values.